minitwit/src/main/java/WebApplication.java

```diff
@@ -206,6 +211,11 @@ public static String getGravatarURL(String email) {
         return gravatar.getUrl(email);
     }
 
+    public static void updateLatest(Request request) {
+        String latest = request.params("latest");
```

Comment:
I thought this only worked to get the params in the url, like /:username/something

Pretty neat if not

Reply:
Honestly, I'm not sure because I can't test it yet. In the Python tests, the data for latest is given as "params" to the POST request. And so I thought it would probably be called the same thing in Spark. But I saw that you can also say request.session().attributes... so who knows.

Reply:
I think latest is suppose to be a query parameter which in Spring is `request.queryParams("latest")`, right now `/api/msgs` returns a `java.lang.NumberFormatException: Cannot parse null string` on line 216 when running `curl -v --header "Authorization: Basic c2ltdWxhdG9yOnN1cGVyX3NhZmUh" http://localhost:8080/api/msgs?latest=1`.

Reply:
Ah, that makes sense. I'll make the change later when I get back. I forgot you could use curl to test it.

... Though I will admit, I'm still not certain what the difference is between parameters, query parameters and attributes. :P Maybe I need a REST refresher.